File: vdp-java/src/test/java/com/visa/vdp/visacheckout/TestGetPaymentData.java
package com.visa.vdp.visacheckout;

import java.util.HashMap;

import org.apache.commons.lang3.StringUtils;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;
import com.visa.vdp.utils.Property;
import com.visa.vdp.utils.VisaProperties;

public class TestGetPaymentData {

    String apiKey;
    VisaAPIClient visaAPIClient;

    @BeforeTest(groups = "visacheckout")
    public void setup() {
        this.visaAPIClient = new VisaAPIClient();
        apiKey = VisaProperties.getProperty(Property.API_KEY);
    }

    @Test(groups = "visacheckout")
    public void testGetPaymentInfo() throws Exception {
        String baseUri = "wallet-services-web/";
        String resourcePath = "payment/data/{callId}";
        resourcePath = StringUtils.replace(resourcePath, "{callId}", VisaProperties.getProperty(Property.CHECKOUT_CALL_ID));

        CloseableHttpResponse response = this.visaAPIClient.doXPayTokenRequest(baseUri, resourcePath, "apikey=" + this.apiKey, "Get Payment Information Test", "", MethodTypes.GET, new HashMap<String, String>());
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
        response.close();
    }
}

File: visa-dev-samplecode-master/vdp-python/visa/test/visacheckout/test_get_payment_data.py
from visa.helpers.visa_api_client import VisaAPIClient
import unittest
import sys
import os
if sys.version_info < (3, 0):
    import ConfigParser as parser
else:
    import configparser as parser
'''
@author: visa
'''

class TestGetPaymentData(unittest.TestCase):
    
    config = parser.ConfigParser()
    config_path = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),'..','configuration.ini'))
    config.read(config_path)
    
    def setUp(self):
        self.visa_api_client = VisaAPIClient()

    def test_get_payment_info(self):
        base_uri = 'wallet-services-web/'
        resource_path = 'payment/data/{callId}'
        resource_path = resource_path.replace('{callId}', self.config.get('VDP','checkoutCallId'))
        query_string = 'apikey=' + self.config.get('VDP','apiKey')
        response = self.visa_api_client.do_x_pay_request(base_uri, resource_path , query_string, '', 'Get Payments Information Test', 'get')
        self.assertEqual(str(response.status_code) ,"200" ,"Get Payments Information test failed")
        pass
